chain/chain/src/store/mod.rs

```diff
@@ -283,10 +272,24 @@ pub trait ChainStoreAccess {
                     // incoming receipts. It would be nicer to explicitly check
                     // that condition rather than relying on errors when reading
                     // from the db.
-                    ret.push(ReceiptProofResponse(current_block_hash, Arc::new(vec![])));
+                    Arc::new(vec![])
                 }
-            }
+            };
+
+            // If the shard layout changed we need to filter receipts to
```

Comment:
So now, is it more like "in some places" we need to filter receipts and "in other places" (state witness) we don't 😄 ? 
 
🤯 🤯

Reply:
removed the comment